javagen/src/main/resources/CoreToCodegenBridgeUtils.java

```diff
@@ -0,0 +1,181 @@
+     */
+    public static ResponseError responseErrorFromJson(JsonReader jsonReader) throws IOException {
+        return jsonReader.readObject(reader -> {
+            // Buffer the next JSON object as ResponseError can take two forms:
+            //
+            // - A ResponseError object
+            // - A ResponseError object wrapped in an "error" node.
+            JsonReader bufferedReader = reader.bufferObject();
+            bufferedReader.nextToken(); // Get to the START_OBJECT token.
+            while (bufferedReader.nextToken() != JsonToken.END_OBJECT) {
+                String fieldName = bufferedReader.getFieldName();
+                bufferedReader.nextToken();
+
+                if ("error".equals(fieldName)) {
+                    // If the ResponseError was wrapped in the "error" node begin reading it now.
+                    return readResponseError(bufferedReader);
+                } else {
+                    bufferedReader.skipChildren();
+                }
+            }
+
+            // Otherwise reset the JsonReader and read the whole JSON object.
+            return readResponseError(bufferedReader.reset());
+        });
+    }
+
+    private static ResponseError readResponseError(JsonReader jsonReader) throws IOException {
+        return jsonReader.readObject(reader -> {
+            String code = null;
+            boolean codeFound = false;
+            String message = null;
+            boolean messageFound = false;
+
+            while (reader.nextToken() != JsonToken.END_OBJECT) {
+                String fieldName = reader.getFieldName();
+                reader.nextToken();
+
+                if ("code".equals(fieldName)) {
+                    code = reader.getString();
+                    codeFound = true;
+                } else if ("message".equals(fieldName)) {
+                    message = reader.getString();
+                    messageFound = true;
+                } else {
+                    reader.skipChildren();
+                }
+            }
+
+            if (!codeFound && !messageFound) {
+                throw new IllegalStateException("Missing required properties: code, message");
+            } else if (!codeFound) {
+                throw new IllegalStateException("Missing required property: code");
+            } else if (!messageFound) {
+                throw new IllegalStateException("Missing required property: message");
+            }
+
+            return new ResponseError(code, message);
+        });
+    }
+
+    /**
+     * Converts a {@link Duration} to a string in ISO-8601 format with support for a day component.
+     *
+     * @param duration The {@link Duration} to convert.
+     * @return The {@link Duration} as a string in ISO-8601 format with support for a day component, or null if the
+     * provided {@link Duration} was null.
+     */
+    public static String durationToStringWithDays(Duration duration) {
```

Comment:
I assume the case of `P24H` is not exactly an error, it can be `P24H` or `P1D`. Here we just make the behavior better?

Reply:
I added this to align with what Jackson serialization was doing. `P24H` and `P1D` should be treated that same as a Duration.

We could decide to remove this and change the test logic to directly compare the duration value and not the String value.